moveit_ros/planning_interface/move_group_interface/src/move_group_interface.cpp

```diff
@@ -1131,16 +1130,22 @@ class MoveGroupInterface::MoveGroupInterfaceImpl
             c.position_constraints.clear();
           if (active_target_ == POSITION)
             c.orientation_constraints.clear();
-          goal.request.goal_constraints[i] =
-              kinematic_constraints::mergeConstraints(goal.request.goal_constraints[i], c);
+          request.goal_constraints[i] = kinematic_constraints::mergeConstraints(request.goal_constraints[i], c);
         }
       }
     }
     else
-      ROS_ERROR_NAMED("move_group_interface", "Unable to construct goal representation");
+      ROS_ERROR_NAMED("move_group_interface", "Unable to construct MotionPlanRequest representation");
 
     if (path_constraints_)
-      goal.request.path_constraints = *path_constraints_;
+      request.path_constraints = *path_constraints_;
+  }
+
+  void constructGoal(moveit_msgs::MoveGroupGoal& goal_out)
+  {
+    // in case something fails during construction, we don't want to overwrite the current goal
```

Comment:
i don't understand why this extra variable needs to be instantiated. ``constructMotionPlanRequest`` cannot return false, so it never fails during construction, and there would be no "overwrite" recovery. your comment is misleading and should be removed along with ``moveit_msgs::MoveGroupGoal goal;``